### ISSUE TITLE
Follow the agreed policy for file and class phpdoc blocks in 1-artifact files

### ISSUE THREAD

**Issue by stronk7** [moodlehq/moodle-local_moodlecheck#66]
This is about to implement the new policy agreed @ [MDLSITE-2804](https://tracker.moodle.org/browse/MDLSITE-2804). Basically:

- In 1-artifact php files (class, interface, trait), normally under `classes` dir, but also out from them.
- Verify that a class phpdoc exists and have all the required elements (package, copyright, license + optional category).
- Ensure that the file phpdoc is optional for those files (but validate it if present).
- Other PHP files will continue requiring a file php doc block.

**Comment by mudrd8mz** [comment]
+1

The plugin still raises "Line 17: File-level phpdocs block is not found" for files that follow the new class-only-block style.

**Comment by stronk7** [comment]
Note I've spent some time today with #69 that doesn't implements this completely, but at least it now stops asking for the file phpdoc block when the file is an 1-artifact one (classes, interfaces and traits).

There are more things to do here, but that's one of the specially annoying ones when new files (usually being 1-artifact files) are added.

Ciao :-)

**Comment by mudrd8mz** [comment]
Thanks Eloy. Maybe that would be enough for now to release a new version and start using it in Github actions? It is sad to see code implemented according to latest policy decisions, causing these failures.

**Comment by stronk7** [comment]
Yeah, I created https://github.com/moodlehq/moodle-plugin-ci/pull/88 some days ago to generate a new release there... will send some friendly pings around...

**Comment by stronk7** [comment]
Update, moodle-plugin-ci 3.0.7 is [being released right now](https://github.com/moodlehq/moodle-plugin-ci/actions/runs/600313533). So jobs out there should start to automatically pick it soon (includes  #69, partial implementation of this).

Ciao :-)

**Comment by timhunt** [comment]
Isn't this now working correctly? Can this be closed?
